js/apps/system/_admin/aardvark/APP/react/src/views/collections/useFetchCollections.ts

```diff
@@ -0,0 +1,56 @@
+import { CollectionMetadata } from "arangojs/collection";
+import { useEffect, useState } from "react";
+import useSWR from "swr";
+import {  getCurrentDB } from "../../utils/arangoClient";
+
+export interface LockableCollectionDescription extends CollectionMetadata {
+  isLocked?: boolean;
+  lockReason?: string;
+}
+
+export const useFetchCollections = () => {
+  const { data, ...rest } = useSWR<LockableCollectionDescription[]>("/collections", () => {
+    return getCurrentDB().listCollections(false);
+  });
+
+  const [collections, setCollections] = useState<LockableCollectionDescription[] | undefined>(
+    data
+  );
+
+  const checkProgress = () => {
+    const callback = (_: any, lockedCollections?: { collection: string; desc: string; }[]) => {
+      const newCollections = data?.map(collection => {
+        const lock = lockedCollections?.find(lockedCollection => lockedCollection.collection === collection.name);
+        if (
+          lock
+        ) {
+          return { ...collection, isLocked: true, lockReason:  lock.desc};
+        }
+        return { ...collection, isLocked: false };
+      });
+      if (newCollections) {
+        setCollections(newCollections);
+      }
+    };
+    if (!window.frontendConfig.ldapEnabled) {
+      window.arangoHelper.syncAndReturnUnfinishedAardvarkJobs("index", callback);
```

Comment:
This is ok now, but we should think of moving this away from the arangoHelper and into a typed function. Same problem in views UI too, so can be done later together